payments-core/src/main/java/com/stripe/android/paymentsheet/AddPaymentMethodsAdapter.kt

```diff
@@ -0,0 +1,94 @@
+package com.stripe.android.paymentsheet
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.stripe.android.R
+import com.stripe.android.databinding.LayoutPaymentsheetAddPaymentMethodCardViewBinding
+import com.stripe.android.paymentsheet.model.SupportedPaymentMethod
+import kotlin.math.roundToInt
+import kotlin.properties.Delegates
+
+internal class AddPaymentMethodsAdapter(
+    private val paymentMethods: List<SupportedPaymentMethod>,
+    val paymentMethodSelectedListener: (paymentMethod: SupportedPaymentMethod) -> Unit
+) : RecyclerView.Adapter<AddPaymentMethodsAdapter.AddPaymentMethodViewHolder>() {
```

Comment:
Hmm, that adapter requires some extra parameters because it adds diffing functionality which we don't need since the payment methods do not change.

Reply:
The idea of the class is you only need to implement the logic to create VH and not worry about maintaining the list passed in. For the diff callback you can just pass in a dummy one to heck the class names, it's used to further optimize RecyclerView when an cell item can be reused to display another data object.

Since this is not a complicated adapter will leave it up to you :)